cookbooks/tomography_based_plate_motions/2D_slice_with_faults_slabs_and_topo.prm

```diff
@@ -0,0 +1,168 @@
+set Start time                             = 0
+set End time                               = 0
+set Max nonlinear iterations               = 30
+set Adiabatic surface temperature          = 1573.0
+
+# We use matrix-free solver and geometric multigrid preconditioner
+# to reduce memory consumption.
+subsection Solver parameters
+  subsection Stokes solver parameters
+    set Linear solver tolerance                         = 1e-4
+    set Stokes solver type                              = block GMG
+    set Number of cheap Stokes solver steps             = 5000
+    set GMRES solver restart length                     = 500
+    set Maximum number of expensive Stokes solver steps = 0
+    set Use full A block as preconditioner              = true
+    set Linear solver A block tolerance                 = 1e-2
+  end
+
+  subsection Diffusion solver parameters
+    set Diffusion length scale = 30000
+  end
+
+end
+
+# The reference profile uses hydrostatic equations to define adiabatic pressure
+# and temperatures. The difference with existing compute profile plugin is that this
+# plugin uses reference densities from PREM below a certain depth defined by the
+# uppermost mantle thickness parameter to compute the adiabatic conditions.
+subsection Adiabatic conditions model
+  set Model name = reference profile
+
+  subsection Reference profile
+    subsection Ascii data model
+      set Data directory = $ASPECT_SOURCE_DIR/data/1D_reference_profiles/
+      set Data file name = prem.txt
+    end
+  end
+end
+
+# We use the spherical shell geometry using the real Earth radius values.
+subsection Geometry model
+  set Model name = spherical shell
+
+  subsection Spherical shell
+    set Inner radius  = 3481000
+    set Outer radius  = 6371000
+  end
+
+  subsection Initial topography model
+    set Model name = ascii data
+
+    subsection Ascii data model
+      set Data directory       = $ASPECT_SOURCE_DIR/cookbooks/tomography_based_plate_motions/input_data/
+      set Data file name       = input_topography_smooth_gauss.txt
+    end
+  end
+
+#  subsection Initial topography model
+#    set Model name = function
+#
+#    subsection Function
+#      set Variable names      = r, phi
+#      set Function constants  = h = 8e3, n_phi = 1
+#      set Coordinate system   = spherical
+#      set Function expression = h*sin(n_phi*phi)
+#    end
+#  end
+end
+
+# We use adaptive refinement to better resolve the plate boundaries and
+# the near-surface heterogeneity
+subsection Mesh refinement
+  set Initial adaptive refinement = 4
+  set Initial global refinement = 4
+  set Strategy = minimum refinement function, composition threshold
+  set Skip solvers on initial refinement = true
+
+  subsection Minimum refinement function
+    set Variable names = depth, y
+    set Function expression = if(depth > 350000, 5, 7 )
+  end
+
+  subsection Composition threshold
+    set Compositional field thresholds = 1e6, 1e6, 1e6, 1e6, 1e6, 0.2
```

Comment:
Per the comment above, was the intention to do the following?
```suggestion
    set Compositional field thresholds = 1e6, 1e6, 1e6, 1e6, 0.2, 1e6
```

Reply:
Thank you for the suggestion. I will fix the comment according to the refinement criterion.